src/sentry/static/sentry/app/views/settings/projectAlerts/issueEditor/index.tsx

```diff
@@ -122,7 +132,80 @@ class IssueRuleEditor extends AsyncView<Props, State> {
     return endpoints as [string, string][];
   }
 
-  handleSubmit = async () => {
+  pollHandler = async (quitTime: number) => {
+    if (Date.now() > quitTime) {
+      addErrorMessage(t('Looking for that channel took too long :('));
+      this.setState({loading: false});
+      return;
+    }
+
+    const {organization, project} = this.props;
+    const {uuid} = this.state;
+    const origRule = {...this.state.rule};
+    let response: RuleTaskResponse = null;
+
+    try {
+      response = await this.api.requestPromise(
+        `/projects/${organization.slug}/${project.slug}/rule-task/${uuid}/`
+      );
+    } catch {
+      addErrorMessage(t('An error occurred'));
+      this.setState({loading: false});
+    }
+
+    if (response) {
+      const {status, rule, error} = response;
+
+      if (status === 'pending') {
+        setTimeout(() => {
+          this.pollHandler(quitTime);
+        }, 1000);
+        return;
+      }
+
+      if (status === 'failed') {
+        this.setState({
+          detailedError: {actions: [error ? error : 'An error occurred']},
+          loading: false,
+        });
+        addErrorMessage(t('An error occurred'));
+      }
+      if (rule && status === 'success') {
+        const ruleId = isSavedAlertRule(origRule) ? `${origRule.id}/` : '';
+        const isNew = !ruleId;
+        this.handleRuleSuccess(isNew, rule);
+      }
+    }
+  };
+
+  fetchStatus() {
+    // pollHander calls itself until it gets either a sucesss
+    // or failed status but we don't want to poll forever so we pass
+    // in a hard stop time of 3 minutes before we bail.
+    const quitTime = Date.now() + POLLING_MAX_TIME_LIMIT;
+    setTimeout(() => {
+      this.pollHandler(quitTime);
+    }, 1000);
+  }
+
+  handleRuleSuccess = (isNew: boolean, rule: IssueAlertRule) => {
+    const {organization} = this.props;
+    this.setState({detailedError: null, loading: false, rule});
+
+    // The onboarding task will be completed on the server side when the alert
+    // is created
+    updateOnboardingTask(null, organization, {
+      task: OnboardingTaskKey.ALERT_RULE,
+      status: 'complete',
+    });
+
+    // When editing, there is an extra route to move back from
+    const stepBack = isNew ? -1 : -2;
+    browserHistory.replace(recreateRoute('', {...this.props, stepBack}));
+    addSuccessMessage(isNew ? t('Created alert rule') : t('Updated alert rule'));
```

Comment:
@MeredithAnya was it your intention to move the `addSuccessMessage` to the end? In the code below, it used to be right after ` updateOnboardingTask`.

Reply:
Might be better this way to have it after the navigation